project_1/include/utils.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <cstdint>
+#include <iostream>
+#include <vector>
+
+enum event_type: std::uint8_t {
```

Comment:
I suggest using `enum class` instead of `enum`. It will help to avoid possible errors with implicit conversion
```suggestion
enum class event_type: std::uint8_t {
```